grpc/server/src/dronecore_server.cpp

```diff
@@ -50,13 +50,15 @@ void RunServer()
     map_type map;
     std::string plugin;
     std::fstream file;
-    file.open("grpc/proto/plugins.conf");
+    file.open("grpc/server/src/plugins/plugins.conf");
     if (!file) {
         LogErr() << "Error in reading conf file";
         return;
     }
     std::vector<::grpc::Service *> list;
-${PLUGIN_MAP_APPEND_STRING}
+    map["action"] = &createInstances<ActionRPCImpl>;
```

Comment:
Any specific reason to again having this hardcoded plugin names ?

Reply:
Right now, the includes are hardcoded anyway, so having both is not consistent to me. I kept the simpler one. But the auto-generation is still in the git history, so we could easily go back to it.

My feeling is that in the end, if we really want plugins, we cannot keep them in-process in dronecore_server. More specifically, if we require the customers to build their own version of dronecore_server in order to use their set of plugins, then it starts to look like a fork (i.e. they will provide their own SDK to their customers). And in this case their are not really "plugins".

So if we want real plugins, they should run in their own process and have a way to register to dronecore_server, other than being detected by cmake and compiled this way. That's probably something we should discuss during the next call :-).

Reply:
>Right now, the includes are hardcoded anyway, so having both is not consistent to me.
For includes also, we can have one more run-time variable like PLUGIN_MAP_APPEND_STRING and assign it in autogenerate_plugin_container.cmake.

Reply:
Yes, definitely.

Still, I'd like to discuss that, as I am not completely sure this is what we want to do in the end. To be honest, I am not a big fan of auto-generating source files, as it adds some magic to the build system.

Again, I believe that either we go towards real plugins, in which case the plugins register themselves to the server, or we go towards some kind of drivers, in which case customers have to compile their own SDK. In case we go for the latter, I'd rather have them understand which library they link and how than having this auto-generated.

I would say we keep this discussion for the next WG call, if you don't mind :-).

Reply:
I vote against compile time code generation as well. It has created quite a bit of confusion in the past and makes it hard to get into this project.

Reply:
I could not agree more! To me, code generation is the same as `#ifdef`s. It is **always** worth spending time doing everything we can for not doing it :D.